nessie/system_test.go

```diff
@@ -129,6 +72,50 @@ func uploadFile(ctx context.Context, t *testing.T, repo, branch, objPath string)
 			WithContent(contentReader), nil)
 
 	require.NoError(t, err, "failed to upload file")
-	payload := stats.GetPayload()
-	return payload.Checksum, objContent
+	return stats.Payload.Checksum, objContent
+}
+
+func listRepositoryObjects(ctx context.Context, t *testing.T, repository string, ref string) []*models.ObjectStats {
+	const amount = 5
+	var entries []*models.ObjectStats
+	var after string
+	for {
+		resp, err := client.Objects.ListObjects(
+			objects.NewListObjectsParamsWithContext(ctx).
+				WithRepository(repository).
+				WithRef(ref).
+				WithAfter(swag.String(after)).
+				WithAmount(swag.Int64(amount)),
+			nil)
+		require.NoError(t, err, "listing objects")
+
+		entries = append(entries, resp.Payload.Results...)
+		after = resp.Payload.Pagination.NextOffset
+		if !swag.BoolValue(resp.Payload.Pagination.HasMore) {
+			break
+		}
+	}
+	return entries
+}
+
+func listRepositories(t *testing.T, ctx context.Context) []string {
```

Comment:
```suggestion
func listRepositoriesIDs(t *testing.T, ctx context.Context) []string {
```

Reply:
Wrap the basic implementation in case a test will require to verify the storage or other information from the repository response